### ISSUE TITLE
compact cnn how to obtain the dominant tags ?

### ISSUE THREAD

**Issue by emmanuelbernieri** [keunwoochoi/music-auto_tagging-keras#32]
Hello again, 

i've implemented the compact-cnn neural network however the output is a vector of 0 & 1 corresponding to inditifed taggs. Is it possible like in the CRNN to only have the dominant taggs like : 

[('jazz', '0.444'), ('instrumental', '0.151'), ('folk', '0.103'), ('Hip-Hop', '0.103'), ('ambient', '0.077')]

Kind regards 

Emmanuel

**Comment by keunwoochoi** [comment]
> however the output is a vector of 0 & 1 corresponding to inditifed taggs

Really? I'm using Sigmoid as an activation function in the output layer, so it can't be like that if you're using the result as it is.

**Comment by emmanuelbernieri** [comment]
Does the music file format has an importance ? I cannot analyse mp3 so I convert them into wav and then I analyse. 

after preprocessing the audio input with librosa it gives me an output like that: 

[[[ 0.          0.          0.         ...,  0.04826343  0.05228163
    0.05081845]]]

**Comment by keunwoochoi** [comment]
Yes, whatever the file format is, it should be decoded which you do when you're using `librosa.load`.

**Comment by emmanuelbernieri** [comment]
even when I use the 1100103.clip.npy. The output of the prediction is still [0.....1]

![model](https://user-images.githubusercontent.com/18503321/30113319-a77d917c-9314-11e7-8b9f-fa8de51a1b7f.PNG)

**Comment by emmanuelbernieri** [comment]
I'm using 

python 3.6 
Keras 2.0.6
Tensorflow 1.3.0
Theano 0.9.0
Librosa 0.5.1

**Comment by keunwoochoi** [comment]
Please check out the keras version (and possibly others) as readme. I think the theano version was either 0.7 or 0.8 at the moment.

**Comment by emmanuelbernieri** [comment]
I'll try to set up exactly the same config as yours tomorrow and let you know the result

**Comment by emmanuelbernieri** [comment]
I didn't succeed in reproducing your result. 

Do you have a docker for a demo ? 

Kind regards 

Emmanuel

**Comment by keunwoochoi** [comment]
> Do you have a docker for a demo ?

No I don’t.

**Comment by emmanuelbernieri** [comment]
thanks for everything i'll let you know when I find where the problem is
